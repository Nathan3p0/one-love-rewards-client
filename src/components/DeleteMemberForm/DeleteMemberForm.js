import React from 'react'
import { Link } from 'react-router-dom'

const DeleteMemberForm = (props) => {
    return (
        <form onSubmit={props.handleSubmit} className="dashboard__add-member--form">
            <h2>Delete Rewards Member</h2>
            <div>
                <label htmlFor="customer-phone">Customer Phone Number:</label>
                <br />
                <input type="tel" id="customer-phone" name="phone_number" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" onChange={props.handleInputChange}
                    placeholder="734-555-0909" required />
            </div>
            <button type="submit" className="create-btn" >Delete User</button>
            <Link to='/dashboard'><button className="go-back-btn">Go Back</button></Link>
        </form>
    )
}

export default DeleteMemberForm