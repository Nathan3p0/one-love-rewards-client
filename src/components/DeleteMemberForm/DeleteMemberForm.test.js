import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import DeleteMemberForm from './DeleteMemberForm'

describe('Member List Item Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<DeleteMemberForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<DeleteMemberForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<DeleteMemberForm
            handleSubmit={() => { }}
            handleInputChange={() => { }}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})