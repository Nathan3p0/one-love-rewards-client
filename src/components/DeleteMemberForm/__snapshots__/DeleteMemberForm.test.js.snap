// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Item Component Rendering Tests renders to UI as expected 1`] = `
<form
  className="dashboard__add-member--form"
  onSubmit={[Function]}
>
  <h2>
    Delete Rewards Member
  </h2>
  <div>
    <label
      htmlFor="customer-phone"
    >
      Customer Phone Number:
    </label>
    <br />
    <input
      id="customer-phone"
      name="phone_number"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="734-555-0909"
      required={true}
      type="tel"
    />
  </div>
  <button
    className="delete-btn"
    type="submit"
  >
    Delete User
  </button>
</form>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<form
  className="dashboard__add-member--form"
  onSubmit={[Function]}
>
  <h2>
    Delete Rewards Member
  </h2>
  <div>
    <label
      htmlFor="customer-phone"
    >
      Customer Phone Number:
    </label>
    <br />
    <input
      id="customer-phone"
      name="phone_number"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="734-555-0909"
      required={true}
      type="tel"
    />
  </div>
  <button
    className="delete-btn"
    type="submit"
  >
    Delete User
  </button>
</form>
`;
